skimage/feature/corner.py

```diff
@@ -1052,18 +1050,12 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=None,
            [2, 3],
            [3, 2],
            [3, 3]])
-    >>> corner_peaks(response, threshold_rel=0)
+    >>> corner_peaks(response)
     array([[2, 2]])
 
     """
     if threshold_rel is None:
         threshold_rel = 0.1
-        warn("Until version 0.16, threshold_rel was set to 0.1 by default. "
-             "Starting from version 0.16, the default value is set to None. "
-             "Until version 0.18, a None value corresponds to a threshold "
-             "value of 0.1. The default behavior will match "
-             "skimage.feature.peak_local_max. To avoid this warning, set "
-             "threshold_rel=0.", category=FutureWarning, stacklevel=2)
```

Comment:
My reading of this is actually that the new default should be 0? Then *that* is what must go in the API change... Right?

Reply:
In the signature, threshold_rel must be None. No modification of this variable must be done in this function. Therefore, the if statement has to be removed. This is to match feature.peak_local_max.

Reply:
So, then, in API changes, we should list, "The default value of threshold_rel in feature.corner has changed from 0.1 to None, which corresponds to letting feature.peak_local_max decide on the default. That is currently equivalent to threshold_rel=0."

Something like that? The point is that the same function call (`corner_peaks(arr)`) will return different things between 0.17 and 0.18. That is what absolutely needs documenting in the release notes.